BuildResidentialHPXML/measure.rb

```diff
@@ -3467,8 +3484,67 @@ def self.set_roofs(hpxml, runner, model, args)
 
   def self.set_rim_joists(hpxml, runner, model, args)
     model.getSurfaces.sort.each do |surface|
-      # TODO
+      next if surface.surfaceType != 'Wall'
+      next unless ['Outdoors', 'Adiabatic'].include? surface.outsideBoundaryCondition
+      next unless Geometry.surface_is_rim_joist(surface, args[:geometry_rim_joist_height])
+
+      interior_adjacent_to = get_adjacent_to(surface)
+      next unless [HPXML::LocationBasementConditioned, HPXML::LocationBasementUnconditioned, HPXML::LocationCrawlspaceUnvented, HPXML::LocationCrawlspaceVented].include? interior_adjacent_to
+
+      exterior_adjacent_to = HPXML::LocationOutside
+      if surface.outsideBoundaryCondition == 'Adiabatic' # can be adjacent to foundation space
+        adjacent_surface = get_adiabatic_adjacent_surface(model, surface)
+        if adjacent_surface.nil? # adjacent to a space that is not explicitly in the model
+          unless [HPXML::ResidentialTypeSFD].include?(args[:geometry_unit_type])
+            exterior_adjacent_to = interior_adjacent_to
+            if exterior_adjacent_to == HPXML::LocationLivingSpace # living adjacent to living
+              exterior_adjacent_to = HPXML::LocationOtherHousingUnit
+            end
+          end
+        else # adjacent to a space that is explicitly in the model, e.g., corridor
+          exterior_adjacent_to = get_adjacent_to(adjacent_surface)
+        end
+      end
+
+      if exterior_adjacent_to == HPXML::LocationOutside && args[:wall_siding_type].is_initialized
+        siding = args[:wall_siding_type].get
+      end
+
+      if args[:wall_color] == Constants.Auto && args[:wall_solar_absorptance] == Constants.Auto
+        solar_absorptance = 0.7
+      end
+
+      if args[:wall_color] != Constants.Auto
+        color = args[:wall_color]
+      end
+
+      if args[:wall_solar_absorptance] != Constants.Auto
+        solar_absorptance = args[:wall_solar_absorptance]
+      end
+
+      if args[:wall_emittance] != Constants.Auto
+        emittance = args[:wall_emittance]
+      end
+
+      hpxml.rim_joists.add(id: valid_attr(surface.name),
+                           exterior_adjacent_to: exterior_adjacent_to,
+                           interior_adjacent_to: interior_adjacent_to,
+                           area: UnitConversions.convert(surface.grossArea, 'm^2', 'ft^2').round(1),
+                           siding: siding,
+                           color: color,
+                           solar_absorptance: solar_absorptance,
+                           emittance: emittance,
+                           insulation_assembly_r_value: args[:rim_joist_assembly_r])
```

Comment:
We should only assign this R-value to thermal boundary rim joists. So do the same thing as we do for walls [here](https://github.com/NREL/OpenStudio-HPXML/blob/87e7a869e84be16b000e16b1cf602f6a2ffe17fc/BuildResidentialHPXML/measure.rb#L3645-L3649) (excluding the `is_uncond_attic_roof_insulated` part, since we're not dealing with attic gable walls).